docs/development/contributing_guide.md

```diff
@@ -114,6 +117,11 @@ Some documentation also exists in [Synapse's GitHub
 Wiki](https://github.com/matrix-org/synapse/wiki), although this is primarily
 contributed to by community authors.
 
+When changes are made to any Rust code then you must call either `poetry install`
+(or `maturin develop` if installed) to rebuild the Rust code. Using [`maturin`](https://github.com/PyO3/maturin)
+is quicker than `poetry install`, so is recommended when making frequent
+changes to the Rust code.
```

Comment:
Maybe worth thinking down the line: is there a way that someone who only wants to touch the Python code can short-cut this setup step / requirement of a rust compiler, e.g. by first installing a wheel?
Certainly not a blocker, just a though

Reply:
There's no good support for doing that TBH. I think we'd have to have the rust as a separate python package to make that work.